File: 06.bean-conditional/src/main/java/study/spring/beanconditional/v1/FalsePrinterConfiguration.java
package study.spring.beanconditional.v1;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import study.spring.beanconditional.v1.model.BooleanPrinterV1;
import study.spring.beanconditional.v1.model.FalsePrinterV1;

@Configuration
@ConditionalOnProperty(value = "application.printer.boolean-printer.v1", havingValue = "false")
public class FalsePrinterConfiguration {

  @Bean
  public BooleanPrinterV1 booleanPrinter() {
    return new FalsePrinterV1();
  }

}

File: bean-conditional/src/main/kotlin/study/spring/beanconditional/FalsePrinterConfiguration.kt
package study.spring.beanconditional

import org.springframework.context.annotation.*
import org.springframework.core.type.AnnotatedTypeMetadata

@Configuration
@Conditional(FalsePrinterCondition::class)
class FalsePrinterConfiguration {

    @Bean
    fun falsePrinter(): BooleanPrinter = FalsePrinter()
}

class FalsePrinterCondition: Condition {
    override fun matches(context: ConditionContext, metadata: AnnotatedTypeMetadata): Boolean {
        return false
    }

}